pkg/mount/volume.go

```diff
@@ -43,29 +43,34 @@ func initRawVolume(ctx context.Context, path string, fsType string) error {
 
 // waitForVolume waits for the StorageOS raw volume to be created.  The retry
 // timeout doubles on every invocation.
-func waitForVolume(path string) error {
+func waitForVolume(ctx context.Context, path string) error {
 
 	var retries int
 	start := time.Now()
 
 	for {
-		timeOff := backoff(retries)
-		if volumeExists(path) {
-			if volumeWritable(path) {
-				// volume ready, exit
-				return nil
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("deadline exceeded while waiting for volume")
+		default:
+			timeOff := backoff(retries)
+			if volumeExists(path) {
+				if volumeWritable(path) {
+					// volume ready, exit
+					return nil
+				}
+				log.Debugf("waiting for volume to come online: %s, retrying in %v", path, timeOff)
+			} else {
+				log.Debugf("waiting for volume: %s, retrying in %v", path, timeOff)
 			}
-			log.Infof("waiting for volume to come online: %s, retrying in %v", path, timeOff)
-		} else {
-			log.Infof("waiting for volume: %s, retrying in %v", path, timeOff)
-		}
 
-		if abort(start, timeOff) {
-			return fmt.Errorf("timed out waiting for volume to be ready")
-		}
+			if abort(start, timeOff) {
+				return fmt.Errorf("timed out waiting for volume to be ready")
+			}
 
-		retries++
-		time.Sleep(timeOff)
+			retries++
+			time.Sleep(timeOff)
```

Comment:
This would cause us to sleep (up to the value of `timeOff`) longer than the context deadline.
However as this is the functionality in other parts of the code-base already I think we should merge.

As a future fix-up we could do something like:
```
var timeOff time.Duration
for {
    select {
    case <-ctx:
        // Do stuff
        return
    case <-timeOff:
        // Do stuff
        // update timeOff
    }
}
```
The initial case of timeOff would be 0 and this case immediately hit